### PR TITLE
Expense Tags

### PR DESCRIPTION
**Pull request by kewitz**
Requires https://github.com/opencollective/opencollective-api/pull/3640
Resolves opencollective/opencollective#2823

Still using the old tag style and input components.

### CODE REVIEW

File: components/StyledSelect.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { typography, layout, space } from 'styled-system';
 import Select, { components } from 'react-select';
 import { injectIntl, defineMessages } from 'react-intl';
 import propTypes from '@styled-system/prop-types';
-import { truncate } from 'lodash';
+import { truncate, unescape } from 'lodash';
```
Comment (kewitz) [inline components/StyledSelect.js:8]: I'm starting to think I should unescape this in the backend, will update this tomorrow.

File: pages/expense.js [hunk 2]
```diff
@@ -372,7 +377,16 @@ class ExpensePage extends React.Component {
                     <FormattedMessage id="Edit" defaultMessage="Edit" />
                   </StyledButton>
                 </Box>
-                <ExpenseInfoSidebar isLoading={data.loading} collective={collective} host={host} />
+                <ExpenseInfoSidebar
+                  isLoading={data.loading}
+                  collective={collective}
+                  host={host}
+                  expense={editedExpense || expense}
```
Comment (Betree) [inline pages/expense.js:384]: **[:sparkles: **Enhancement**]** I would do:

```suggestion
                  expense={status === PAGE_STATUS.VIEW ? expense : editedExpense}
```

It's very minor, but the reason is that `editedExpense` can get out of sync. `expense` is refreshed automatically from the caches, and thanks to the `pollInterval`.

File: components/expenses/PayoutBankInformationForm.js [hunk 3]
```diff
@@ -45,7 +45,7 @@ const FormFieldsContainer = styled(Flex)`
 `;
 
 const requiredFieldsQuery = gqlV2`
-  query Collective($slug: String, $currency: TransferWiseCurrency!) {
+  query Collective($slug: String, $currency: String!) {
```
Comment (Betree) [inline components/expenses/PayoutBankInformationForm.js:48]: Is this change intended?

Comment (kewitz) [reply components/expenses/PayoutBankInformationForm.js:48]: Yes, otherwise any currency update in TransferWise will cause errors on our side.

File: pages/ExpenseInfoSidebar.js [hunk 4]
```diff
@@ -2,22 +2,56 @@ import { Box } from '@rebass/grid';
...
+    display: inline-block;
+  }
+`;
 
 /**
  * Provide some info (ie. collective balance, tags, policies, etc.) for the expense pages
  * in a sidebar.
  */
-const ExpenseInfoSidebar = ({ isLoading, host, collective }) => {
+const ExpenseInfoSidebar = ({ isEditing, isLoading, host, collective, expense, onChange }) => {
```
[the remark this answers is not in the document]
Comment (kewitz) [reply pages/ExpenseInfoSidebar.js:28]: 👍

File: pages/ExpenseInfoSidebar.js [hunk 5]
```diff
@@ -2,22 +2,56 @@ import { Box } from '@rebass/grid';
...
+      <Box>
+        <H5 mb={3}>
+          <FormattedMessage id="Tags" defaultMessage="Tags" />
+        </H5>
+        {isLoading && <LoadingPlaceholder height={28} width="100%" />}
+        {!isLoading &&
+          (isEditing ? (
+            <StyledInputTags onChange={handleTagUpdate} value={expense.tags} />
+          ) : (
+            <TagsWrapper>
```
[the remark this answers is not in the document]
Comment (kewitz) [reply pages/ExpenseInfoSidebar.js:45]: 🥇 This makes me so happy!

**Comment by kewitz** [comment]
Fixed and rebased.

File: pages/ExpenseInfoSidebar.js [hunk 6]
```diff
@@ -1,23 +1,49 @@
...
+      <Box>
+        <H5 mb={3}>
+          <FormattedMessage id="Tags" defaultMessage="Tags" />
+        </H5>
+        {isLoading && <LoadingPlaceholder height={28} width="100%" />}
+        {!isLoading &&
+          (isEditing ? (
+            <StyledInputTags onChange={handleTagUpdate} value={expense.tags} />
+          ) : (
+            <Flex>
```
Comment (Betree) [inline pages/ExpenseInfoSidebar.js:37]: If it breaks with multiple tags, you can try

```suggestion
            <Flex flexWrap="wrap">
```
